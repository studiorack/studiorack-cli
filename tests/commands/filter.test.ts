import { expect, test } from 'vitest';
import { cli } from '../shared';
import { License, RegistryType } from '@open-audio-stack/core';

test('Filter packages', async () => {
  expect(cli(RegistryType.Plugins, 'filter', 'name', 'Surge XT')).toMatchSnapshot();
  expect(cli(RegistryType.Plugins, 'filter', 'name', 'Surge X')).toMatchSnapshot();
  expect(cli(RegistryType.Plugins, 'filter', 'license', License.CreativeCommonsZerov1Universal)).toMatchSnapshot();
});
