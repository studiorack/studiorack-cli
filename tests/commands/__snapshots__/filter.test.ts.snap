// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter packages 1`] = `
"[90m┌─────────────────────────[39m[90m┬──────────[39m[90m┬─────────[39m[90m┬────────────[39m[90m┬─────────[39m[90m┬───────────────────────────────┐[39m
[90m│[39m[31m Id                      [39m[90m│[39m[31m Name     [39m[90m│[39m[31m Version [39m[90m│[39m[31m Date       [39m[90m│[39m[31m License [39m[90m│[39m[31m Tags                          [39m[90m│[39m
[90m├─────────────────────────[39m[90m┼──────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────┤[39m
[90m│[39m surge-synthesizer/surge [90m│[39m Surge XT [90m│[39m 1.3.4   [90m│[39m 2024-08-11 [90m│[39m gpl-3.0 [90m│[39m Instrument, Synth, Modulation [90m│[39m
[90m└─────────────────────────[39m[90m┴──────────[39m[90m┴─────────[39m[90m┴────────────[39m[90m┴─────────[39m[90m┴───────────────────────────────┘[39m"
`;

exports[`Filter packages 2`] = `"No results found"`;

exports[`Filter packages 3`] = `
"[90m┌────────────────────────────────────────[39m[90m┬──────────────────────────[39m[90m┬─────────[39m[90m┬────────────[39m[90m┬─────────[39m[90m┬───────────────────────────────────┐[39m
[90m│[39m[31m Id                                     [39m[90m│[39m[31m Name                     [39m[90m│[39m[31m Version [39m[90m│[39m[31m Date       [39m[90m│[39m[31m License [39m[90m│[39m[31m Tags                              [39m[90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/virtuosity_drums        [90m│[39m Virtuosity Drums         [90m│[39m 0.9.24  [90m│[39m 2022-09-21 [90m│[39m cc0-1.0 [90m│[39m Instrument, Drums, Jazz, sfz      [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/hungarian-zither        [90m│[39m Hungarian Zither         [90m│[39m 1.0.1   [90m│[39m 2022-09-28 [90m│[39m cc0-1.0 [90m│[39m Instrument, Strings, Classical... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/horse-pulse             [90m│[39m Horse Pulse              [90m│[39m 1.0.0   [90m│[39m 2022-04-01 [90m│[39m cc0-1.0 [90m│[39m Instrument, String, Pluck, sfz    [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/double-bass             [90m│[39m Double Bass              [90m│[39m 1.0.1   [90m│[39m 2022-09-04 [90m│[39m cc0-1.0 [90m│[39m Instrument, String, Bass, sfz     [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/black-and-green-guitars [90m│[39m Black and Green Guitars  [90m│[39m 1.0.0   [90m│[39m 2022-08-08 [90m│[39m cc0-1.0 [90m│[39m Instrument, Guitar, Electric, ... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m sfzinstruments/bear-sax                [90m│[39m Bear Sax                 [90m│[39m 1.0.4   [90m│[39m 2021-09-11 [90m│[39m cc0-1.0 [90m│[39m Instrument, Brass, Saxophone, ... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m michael02022/basic-harmonica           [90m│[39m Basic Harmonica          [90m│[39m 1.0.0   [90m│[39m 2022-10-07 [90m│[39m cc0-1.0 [90m│[39m Instrument, Wind, Harmonica, O... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m freepats/upright-piano-kw              [90m│[39m Upright Piano KW         [90m│[39m 1.0.0   [90m│[39m 2022-02-21 [90m│[39m cc0-1.0 [90m│[39m Instrument, Keys, Piano, sfz      [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m freepats/synthesizer-percussion        [90m│[39m Synthesizer Percussion   [90m│[39m 1.0.0   [90m│[39m 2022-07-18 [90m│[39m cc0-1.0 [90m│[39m Instrument, Drums, Electro, sf... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m freepats/spanish-classical-guitar      [90m│[39m Spanish Classical Guitar [90m│[39m 1.0.0   [90m│[39m 2019-06-18 [90m│[39m cc0-1.0 [90m│[39m Instrument, Guitar, Classical,... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m freepats/hang-d-minor                  [90m│[39m Hang D Minor             [90m│[39m 1.0.0   [90m│[39m 2022-03-30 [90m│[39m cc0-1.0 [90m│[39m Instrument, Orchestra, Hang, s... [90m│[39m
[90m├────────────────────────────────────────[39m[90m┼──────────────────────────[39m[90m┼─────────[39m[90m┼────────────[39m[90m┼─────────[39m[90m┼───────────────────────────────────┤[39m
[90m│[39m freepats/glasses-of-water              [90m│[39m Glasses Of Water         [90m│[39m 1.0.0   [90m│[39m 2019-12-27 [90m│[39m cc0-1.0 [90m│[39m Instrument, Orchestra, Glass, ... [90m│[39m
[90m└────────────────────────────────────────[39m[90m┴──────────────────────────[39m[90m┴─────────[39m[90m┴────────────[39m[90m┴─────────[39m[90m┴───────────────────────────────────┘[39m"
`;
